### PR TITLE
Admin add course 

### PR DESCRIPTION
**Pull request by mohdzakwan99**
-register course 
-view course offer 
-database association (has many through)

### CODE REVIEW

**Comment by mohdzakwan99** [comment]
ready

File: app/assets/stylesheets/registers.scss [hunk 1]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the Registers controller here.
```
Comment (ghost) [inline app/assets/stylesheets/registers.scss]: delete this file please

File: app/helpers/course_helper.rb [hunk 2]
```diff
@@ -0,0 +1,2 @@
+module CourseHelper
```
Comment (ghost) [inline app/helpers/course_helper.rb]: delete this file please

File: app/helpers/registers_helper.rb [hunk 3]
```diff
@@ -0,0 +1,2 @@
+module RegistersHelper
```
Comment (anlvq-1058) [inline app/helpers/registers_helper.rb:1]: delete this file please

File: app/models/course.rb [hunk 4]
```diff
@@ -0,0 +1,2 @@
+class Course < ApplicationRecord
```
Comment (anlvq-1058) [inline app/models/course.rb:1]: delete this file please

File: app/models/register.rb [hunk 5]
```diff
@@ -0,0 +1,6 @@
+class Register < ApplicationRecord
+  belongs_to :user
+  scope :newest, -> { order(created_at: :desc) }
+  validates :user_id, presence: true
+  validates :courseid, presence: true, length: {maximum: 8, minimum: 8}
```
Comment (ghost) [inline app/models/register.rb]:
```suggestion
```

File: app/views/course/index.html.erb [hunk 6]
```diff
@@ -0,0 +1,13 @@
+<h1>Course offered</h1>
+
+<ul>
+  <% @course.each do |course| %>
+    <li>
+      <a href="/course/<%= course.id %>">
```
Comment (ghost) [inline app/views/course/index.html.erb]: use link_to please

File: app/views/course/index.html.erb [hunk 7]
```diff
@@ -0,0 +1,13 @@
...
+        <%= course.title %>
+      </a>
+    </li>
+  <% end %>
+</ul>
+
+
```
Comment (ghost) [inline app/views/course/index.html.erb]: remove extend line

File: test/controllers/course_controller_test.rb [hunk 8]
```diff
@@ -0,0 +1,7 @@
+require "test_helper"
+
```
Comment (ghost) [inline test/controllers/course_controller_test.rb]: delete this file please

File: test/controllers/registers_controller_test.rb [hunk 9]
```diff
@@ -0,0 +1,7 @@
+require "test_helper"
+
```
Comment (ghost) [inline test/controllers/registers_controller_test.rb]: delete this file please

File: test/fixtures/courses.yml [hunk 10]
```diff
@@ -0,0 +1,7 @@
+# Read about fixtures at https://api.rubyonrails.org/classes/ActiveRecord/FixtureSet.html
+
+one:
```
Comment (ghost) [inline test/fixtures/courses.yml]: delete this file please

File: test/fixtures/registers.yml [hunk 11]
```diff
@@ -0,0 +1,9 @@
+# Read about fixtures at https://api.rubyonrails.org/classes/ActiveRecord/FixtureSet.html
+
+one:
```
Comment (ghost) [inline test/fixtures/registers.yml]: delete this file please

File: test/models/course_test.rb [hunk 12]
```diff
@@ -0,0 +1,7 @@
+require "test_helper"
+
```
Comment (ghost) [inline test/models/course_test.rb]: delete this file please

File: test/models/register_test.rb [hunk 13]
```diff
@@ -0,0 +1,7 @@
+require "test_helper"
+
+class RegisterTest < ActiveSupport::TestCase
```
Comment (ghost) [inline test/models/register_test.rb]: delete this file please

**Review by ghost — COMMENTED** [review]
cmt

**Comment by mohdzakwan99** [comment]
ready

File: app/helpers/register_helper.rb [hunk 14]
```diff
@@ -0,0 +1,2 @@
+module RegisterHelper
```
Comment (ngocvt-0484) [inline app/helpers/register_helper.rb]: delete this file because not use

File: app/assets/stylesheets/register.scss [hunk 15]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the Register controller here.
+// They will automatically be included in application.css.
```
Comment (ngocvt-0484) [inline app/assets/stylesheets/register.scss]: delete this file because not use

File: app/models/user.rb [hunk 16]
```diff
@@ -8,8 +8,12 @@ class User < ApplicationRecord
   VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i
   validates :email, presence: true, length: { maximum: 255 }, format: { with: VALID_EMAIL_REGEX }
   validates :password, presence: true, length: {minimum: 6}, allow_nil: true
+  scope :newest, -> { order(created_at: :desc) }
   has_secure_password
 
+  has_many :user_courses
+  has_many :course, through: :user_courses
```
Comment (ngocvt-0484) [inline app/models/user.rb]:
```suggestion
  has_many :courses, through: :user_courses
```

File: app/views/register/index.html.erb [hunk 17]
```diff
@@ -0,0 +1,16 @@
...
+  <% @courses.each do |course| %>
+    <li>
+      <%= course.name %><br>
+    </li>
+    <p>
+      <%= t(:description) %> <%= course.description %><br>
+      <%= t(:started_at) %> <%= course.started_at %><br>
+      <%= t(:total_month) %> <%= course.total_month %><br>
+      <%= t(:total_member) %> <%= course.total_member %><br>
+      <%= t(:status) %> <%= course.status%>
```
Comment (ngocvt-0484) [inline app/views/register/index.html.erb]:
```suggestion
      <%= t(:status) %> <%= course.status %>
```

File: app/views/register/new.html.erb [hunk 18]
```diff
@@ -0,0 +1,37 @@
...
+            <%= f.submit "Create course", class: "btn btn-primary" %>
+          <% end %>
+        </div>
+      </div>
+    </body>
+  </html>
+<% else %>
+  <h3>You are not the admin user!</h3>
+<% end %>
+
```
Comment (ngocvt-0484) [inline app/views/register/new.html.erb]: remove blank line

File: app/views/static_pages/home.html.erb [hunk 19]
```diff
@@ -2,5 +2,11 @@
...
 <p>
-  <%= t(:t_p1) %>
+  <%= t(:t_p1) %><br>
+  <% if current_user && current_user.admin? %>
+    <li><%= link_to t(:add_course), register_path %></li>
+  <% else %>
+  <% end %>
 </p>
+
+
```
Comment (ngocvt-0484) [inline app/views/static_pages/home.html.erb]: remove blank line

File: app/views/static_pages/home.html.erb [hunk 20]
```diff
@@ -2,5 +2,11 @@
 
 <h1><%= t(:greetings) %></h1>
 <p>
-  <%= t(:t_p1) %>
+  <%= t(:t_p1) %><br>
+  <% if current_user && current_user.admin? %>
```
Comment (ngocvt-0484) [inline app/views/static_pages/home.html.erb]:
```suggestion
  <% if current_user&.admin? %>
```

**Comment by mohdzakwan99** [comment]
ready

File: test/controllers/register_controller_test.rb [hunk 21]
```diff
@@ -0,0 +1,8 @@
+require "test_helper"
```
Comment (huongnt-2545) [inline test/controllers/register_controller_test.rb]: delete this file, it's not useful

File: test/fixtures/user_courses.yml [hunk 22]
```diff
@@ -0,0 +1,11 @@
+# Read about fixtures at https://api.rubyonrails.org/classes/ActiveRecord/FixtureSet.html
```
Comment (huongnt-2545) [inline test/fixtures/user_courses.yml]: delete this file

File: config/routes.rb [hunk 23]
```diff
@@ -9,6 +9,9 @@
     delete "/logout", to: "sessions#destroy"
     get 'password_resets/new'
     get 'password_resets/edit'
+    get "/register", to: "register#new"
```
Comment (huongnt-2545) [inline config/routes.rb]: line 12 -> 14, you should define the clear router name for course: `resources :courses`

**Comment by mohdzakwan99** [comment]
ready

File: app/controllers/courses_controller.rb [hunk 24]
```diff
@@ -0,0 +1,38 @@
...
+
+  def new
+    @course = Course.new
+  end
+
+  def create
+    @course = Course.new(course_params)
+    if @course.save
+      flash[:success] = "Course successfully registered"
+      redirect_to current_user
```
Comment (huongnt-2545) [inline app/controllers/courses_controller.rb]: should be redirect to course detail or list of courses
```suggestion
      redirect_to courses_url
```

File: app/views/new.html.erb [hunk 25]
```diff
@@ -0,0 +1,36 @@
+<% if current_user.admin %>
```
Comment (huongnt-2545) [inline app/views/new.html.erb]: what's purpose of this file??

Comment (mohdzakwan99) [reply app/views/new.html.erb]: sorry, mistakenly pasted it there

**Comment by mohdzakwan99** [comment]
ready

**Review by ghost — COMMENTED** [review]
merged

**Pull request merged**